src/pages/Vote.vue

```diff
@@ -40,33 +40,90 @@
 
 <script lang="ts">
 import { VALIDATE_MESSAGES } from "@/constants/constant";
+import DaoID from "@/store/DaoID";
+import WalletStore from "@/store/WalletStore";
+import { types } from "@algo-builder/web";
+import { LogicSigAccount } from "algosdk";
 import { defineComponent, reactive } from "vue";
 import VoteStore from "../store/VoteStore";
-import { VoteOptions } from "../types/enum.types";
+import { DAOActions, VoteOptions } from "../types/enum.types";
+import { getProposalLsig } from "../contract/dao";
 
 export default defineComponent({
 	name: "AddProposal",
 	data() {
 		return {
 			VoteOptions,
+			error: "",
 		};
 	},
-	methods: {
-		onFinish(values: Event) {
-			console.log("Success:", values);
-		},
-		onFinishFailed(errorinfo: Event) {
-			console.warn("Failed:", errorinfo);
-		},
-	},
 	setup() {
 		const formState = reactive(VoteStore());
-
+		const daoIDStore = reactive(DaoID());
+		const walletStore = reactive(WalletStore());
 		return {
 			formState,
+			daoIDStore,
+			walletStore,
 			validateMessages: VALIDATE_MESSAGES,
 		};
 	},
+	methods: {
+		async onFinish() {
+			if (typeof this.daoIDStore.dao_id === "undefined") {
+				this.error = "Please add DAO App ID";
+				setTimeout(() => {
+					this.error = "";
+				}, 2000);
+				return;
+			}
+			if (typeof this.daoIDStore.govt_id === "undefined") {
+				this.error = "Govt token not found";
+				setTimeout(() => {
+					this.error = "";
+				}, 2000);
+				return;
+			}
+			if (!this.walletStore.address.length) {
+				this.error = "Please connect to your Wallet";
+				setTimeout(() => {
+					this.error = "";
+				}, 2000);
+				return;
+			}
+			let lsig: LogicSigAccount = await getProposalLsig(
+				this.daoIDStore.dao_id,
+				this.walletStore.address
+			);
+
+			console.log(`* Register votes by ${this.walletStore.address} *`);
+			// call to DAO app by voter (to register deposited votes)
+			const registerVoteParam: types.ExecParams = {
+				type: types.TransactionType.CallApp,
+				sign: types.SignType.SecretKey,
+				fromAccount: {
+					addr: this.walletStore.address,
+					sk: new Uint8Array(0),
+				},
+				appID: this.daoIDStore.dao_id,
+				payFlags: { totalFee: 2000 },
+				appArgs: [DAOActions.REGISTER_VOTE, `str:${this.formState.vote_type}`],
```

Comment:
When I run this branch in my laptop, this.formState.vote_type return number type not string. It should be VoteOptions type(yes,no,abstain)

Please check it!